app/controllers/orders_controller.rb

```diff
@@ -110,7 +115,12 @@ def add
         rescue NUCore::MixedFacilityCart
           @order.errors.add(:base, "You can not add a product from another facility; please clear your cart or place a separate order.")
         rescue => e
-          @order.errors.add(:base, "An error was encountered while adding the product #{@product}.")
+          if e.message == "Validation failed: Account is not valid for the orderer"
+            @order.errors.add(:base, e.message)
+            @order.errors.add(:base, "Please Change Payment Source or remove any invalid products from the Cart before continuing")
```

Comment:
Then we can reuse in 3 places we need it like:
```suggestion
    @order.errors.add(:base, t("account_invalid_for_orderer"))
```